### PR TITLE
support for gcp dnsendpoints

### PR DESCRIPTION
**Pull request by staebler**

### CODE REVIEW

File: pkg/controller/dnsendpoint/nameserver/gcp.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+	}
+	currentNameServers, err := q.queryNameServers(gcpClient, zoneName)
+	return currentNameServers, errors.Wrap(err, "error querying name servers")
+}
+
+// Create implements Query.Create.
+func (q *gcpQuery) Create(rootDomain string, domain string, values sets.String) error {
+	gcpClient, err := q.getGCPClient()
+	if err != nil {
+		return errors.Wrap(err, "failed to get AWS client")
```
Comment (twiest) [inline pkg/controller/dnsendpoint/nameserver/gcp.go]: "failed to get GCP client"

File: pkg/controller/dnsendpoint/nameserver/gcp.go [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+		q.createNameServers(gcpClient, zoneName, domain, values),
+		"error creating the name server",
+	)
+}
+
+// Delete implements Query.Delete.
+func (q *gcpQuery) Delete(rootDomain string, domain string, values sets.String) error {
+	gcpClient, err := q.getGCPClient()
+	if err != nil {
+		return errors.Wrap(err, "failed to get AWS client")
```
Comment (twiest) [inline pkg/controller/dnsendpoint/nameserver/gcp.go]: "failed to get GCP client"

**Comment by staebler** [comment]
f5adc9db...8d28a1b7 fixes the reference to AWS in the GCP files.

```console
$ grep pkg/controller/dnsendpoint/nameserver/gcp* -e AWS -c
pkg/controller/dnsendpoint/nameserver/gcp.go:0
pkg/controller/dnsendpoint/nameserver/gcp_live_test.go:0
pkg/controller/dnsendpoint/nameserver/gcp_test.go:0
```

**Comment by dgoodwin** [comment]
/test e2e

@twiest is this one ready to go?

**Pull request merged**